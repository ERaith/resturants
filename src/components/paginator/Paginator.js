export function Paginator({size }) {
  return (
    <div className="paginator">
      {Array(size)
        .fill(0)
        .map((_, i) => {
          return (
            <button
              className={`paginator-block`}
              id={`page-${i}`}
              key={i}
            >
              {i}
            </button>
          );
        })}
    </div>
  );
}
