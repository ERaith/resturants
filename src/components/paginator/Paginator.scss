.paginator-block {
  @include button-bg(#3498db);
}
