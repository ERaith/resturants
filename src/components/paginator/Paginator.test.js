import React from "react";
import { Paginator } from "./Paginator";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("SelectionFilter", () => {
  describe("Snapshots", () => {
    it("should match the snapshot", () => {
      const mockSetPage = jest.fn();
      const props = {
        page: 0,
        setPage: mockSetPage,
        size: 10,
      };
      const wrapper = shallow(<Paginator {...props} />);

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe("user interactions", () => {
    it("should start with button 0 as active", () => {
      const mockSetPage = jest.fn();
      const props = {
        page: 0,
        setPage: mockSetPage,
        size: 5,
      };
      const wrapper = shallow(<Paginator {...props} />);
      const currentPageBtn = wrapper.find(".active");

      expect(currentPageBtn).toBeTruthy();
      expect(currentPageBtn.text()).toMatch("0");
      expect(wrapper.find(".active").hasClass("paginator-block")).toEqual(true);
    });
    it("should update active class", () => {
      const mockSetPage = jest.fn();
      const props = {
        page: 0,
        setPage: mockSetPage,
        size: 5,
      };
      const wrapper = shallow(<Paginator {...props} />);
      const currentPageBtn = wrapper.find(".active");

      expect(currentPageBtn).toBeTruthy();
      expect(currentPageBtn.text()).toMatch("0");

      const newActivePageBtn = wrapper.find("#page-1");
      newActivePageBtn.simulate("click");

      expect(mockSetPage).toHaveBeenCalledWith(1);
    });
  });
});
