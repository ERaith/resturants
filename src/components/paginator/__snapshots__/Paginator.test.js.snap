// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectionFilter Snapshots should match the snapshot 1`] = `
<div
  className="paginator"
>
  <button
    className="paginator-block active"
    id="page-0"
    key="0"
    onClick={[Function]}
  >
    0
  </button>
  <button
    className="paginator-block "
    id="page-1"
    key="1"
    onClick={[Function]}
  >
    1
  </button>
  <button
    className="paginator-block "
    id="page-2"
    key="2"
    onClick={[Function]}
  >
    2
  </button>
  <button
    className="paginator-block "
    id="page-3"
    key="3"
    onClick={[Function]}
  >
    3
  </button>
  <button
    className="paginator-block "
    id="page-4"
    key="4"
    onClick={[Function]}
  >
    4
  </button>
  <button
    className="paginator-block "
    id="page-5"
    key="5"
    onClick={[Function]}
  >
    5
  </button>
  <button
    className="paginator-block "
    id="page-6"
    key="6"
    onClick={[Function]}
  >
    6
  </button>
  <button
    className="paginator-block "
    id="page-7"
    key="7"
    onClick={[Function]}
  >
    7
  </button>
  <button
    className="paginator-block "
    id="page-8"
    key="8"
    onClick={[Function]}
  >
    8
  </button>
  <button
    className="paginator-block "
    id="page-9"
    key="9"
    onClick={[Function]}
  >
    9
  </button>
</div>
`;
